docs/Building_Diagrams/data_stores.md

```diff
@@ -10,3 +10,141 @@ It can represent a database, file system, or any other storage mechanism.
+
+- **Use Cases**: Ideal for scenarios requiring rich, structured data access within workflows, such as managing inventories, categorizing user profiles, or handling complex hierarchical data like the provided movies example.
+
+#### BPMN Integration
+
+Integrating KKV data stores within BPMN diagrams involves modeling tasks that interact with the data store for CRUD (Create, Read, Update, Delete) operations. 
+
+Here's how to depict such interactions using a BPMN example focused on movie data management:
+
+### BPMN Example: Movie Data Management
+
+#### Process Overview
+![KKV data_store](images/DataStore_KKV.png)
+
+1. **Start Event**: Initiates the movie data management workflow.
+
+2. **Script Task: Populate Data Store**
+    - Uses a script to populate the "Movies" data store with detailed information about characters in various movies, demonstrating the KKV structure (Director -> Movie -> Character Attributes).
+
+3. **Script Task: Extract Single Record**
+    - Retrieves specific movie details from the "Movies" data store.
+    - A script extracts information about "Django Unchained" using the KKV path, illustrating how to access nested data.
+    - **Script**: `django = movies["Quentin Tarantino"]["Django Unchained"]`
+
+5. **Script Task: Insert A Single Record**
+    - Adds a new movie entry to the data store under a different key.
+    - Demonstrates adding an entry for a Wes Craven movie to the "Movies" data store, showcasing how new data can be structured and inserted.
+    - **Script**:
+     ```python
+     movies["Wes Craven"]["Swamp Thing"] = [
+       {"name": "Dr. Alec Hollard", "actor": "Ray Wise", "description": "whatever"}
+     ]
+     ```
+
+6. **End Event**: Marks the completion of the workflow.
+After running the process, You can view the new movies data in data store:
+![KKV data_store](images/DataStore_KKV_Store.png)
+
+#### Modeling Data Store Interactions
+
+- **Data Store Reference**: Each script task interacts with a "Movies" data store, specified in the task's properties. The data store's KKV nature is indicated, allowing for structured data access and manipulation.
+  
+- **Selection of Data Source**: In the properties section of the tasks, the "Movies" data store is selected as the datasource, highlighting how BPMN tasks can be explicitly linked to specific data stores for operations.
+
+### JSON Data Store
+
+In BPMN (Business Process Model and Notation), incorporating a JSON file as a data store offers a versatile method for managing structured data throughout a process. A structured file or set of files that uses the JSON format to store data in a hierarchical key-value pair system. This approach is particularly beneficial for BPMN processes requiring access to configurable, structured data without needing a connection to a database.
```

Comment:
Small nit-pick, but the JSON data doesn't have to be in a key-value pair format, it could be `[1, 2, 3]` for `false` for instance - anything that is a valid JSON value.